flang/lib/Parser/openacc-parsers.cpp

```diff
@@ -225,7 +229,8 @@ TYPE_PARSER(startAccLine >> sourced(construct<AccEndBlockDirective>("END"_tok >>
 
 TYPE_PARSER(construct<OpenACCBlockConstruct>(
     Parser<AccBeginBlockDirective>{} / endAccLine, block,
-    Parser<AccEndBlockDirective>{} / endAccLine))
+    withMessage("expected OpenACC end block directive"_err_en_US,
```

Comment:
still needs error recovery here, I think

Reply:
What case is the error recovery here suppose to improve? I played around with it a little and couldn't get any improvement. My understanding would be the recovery code gets back to a know good state. I think the recovery code in executionPart already does this, but I am not certain.

Reply:
Without explicit error recovery here, a missing `AccEndBlockDirective` will cause the whole `OpenACCBlockConstruct` production to fail.  It might be better to recover from a missing `AccEndBlockDirective` by complaining that it was missing, and then still succeeding on the whole production.

The explicit error recovery could also go into the parser for `AccEndBlockDirective`.